app/models/grouping.rb

```diff
@@ -664,18 +668,11 @@ def section
   ##
   def past_due_date?
     timestamp = assignment_folder_last_modified_date
-    due_dates = assignment.section_due_dates
-    section = unless inviter.blank?
-                inviter.section
-              end
-    section_due_date = unless section.blank? || due_dates.blank?
-                         due_dates.where(section_id: section).first.due_date
-                       end
-
-    if !section_due_date.blank?
-      timestamp > section_due_date
-    else
+    if inviter.blank? || inviter.section.blank? || assignment.section_due_dates.blank?
       timestamp > assignment.due_date
+    else
+      section_due_date = assignment.section_due_dates.where(section_id: inviter.section).first.due_date
```

Comment:
Using `find` instead of `.where().first`

Reply:
Right, I copied and pasted the previous code. Will change it.